Dockerfile.apiserver

```diff
@@ -1,21 +1,27 @@
-# syntax=docker/dockerfile:1.2
+# syntax=docker/dockerfile:1
+
 # xx is a helper for cross-compilation
 FROM --platform=$BUILDPLATFORM tonistiigi/xx:1.3.0@sha256:904fe94f236d36d65aeb5a2462f88f2c537b8360475f6342e7599194f291fb7e AS xx
 
 FROM --platform=$BUILDPLATFORM golang:1.21.3-bullseye AS builder
 
-COPY --from=xx / /
+COPY --link --from=xx / /
 
 ARG TARGETPLATFORM
 
-RUN apt-get update && xx-apt-get install -y --no-install-recommends gcc libc6-dev
-
-RUN xx-go --wrap
+RUN --mount=type=cache,id=${TARGETPLATFORM}-apt,target=/var/cache/apt,sharing=locked \
+    apt-get update \
+    && xx-apt-get install -y --no-install-recommends \
+      gcc \
+      libc6-dev
 
-# Copy vmclarity code to /build and move to that directory
-COPY . /build
 WORKDIR /build
 
+RUN --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=bind,source=.,target=/build,ro \
+    xx-go mod download -x
```

Comment:
@Tehsmash The reason I prefer to use bind mount the source is to avoid invalidating cache when the `go.mod` or `go.sum` files are updated.

Reply:
I'm not taking about invalidating the cache, I'm taking about invalidating the layer. If you don't invalidate the layer then it will never re-run unless you manually force it to rerun which is a bad experience when developing locally because you'll make a change to a dependency and then be very confused why the go mod stage never re-ran.

Reply:
@Tehsmash My understanding is that the cache is being updated (including the layer) on every run.

It seems others do this too: https://github.com/moby/swarmkit/blob/master/Dockerfile